cf_units/tests/unit/unit/test_as_unit.py

```diff
@@ -0,0 +1,87 @@
+# (C) British Crown Copyright 2014, Met Office
+#
+# This file is part of cf_units.
+#
+# cf_units is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cf_units is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with cf_units.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for the `cf_units.as_unit` function."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+import copy
+import unittest
+
+import mock
+
+import cf_units
+from cf_units import as_unit, Unit
+
+
+class StubUnit(object):
```

Comment:
The intention of this bugfix is to fix the case of **non** `cf_units.Unit` objects.  Making use of `cf_units.Unit` would simply return itself and would not ensure that SciTools/iris#1874 has been rectified:

```Python
if isinstance(unit, Unit):
    result = unit
else:
    ...
```
[see](https://github.com/SciTools/cf_units/blob/24b8774fa4e4d1502dfed63812c04db64dd84be9/cf_units/__init__.py#L776)